src/openforms/js/components/admin/form_design/logic/Trigger.js

```diff
@@ -90,10 +90,17 @@ const parseJsonLogic = (logic) => {
     }
 
     // check if we're using a literal value, or a component reference
-    const compareValue = values[1];
+    let compareValue = values[1];
     let operandType = '';
     let operand = '';
 
+    // Selectboxes case: the component name contains the reference to which value, e.g. "selectComponentField.option1"
+    const splitComponent = component.split('.');
+    if (splitComponent.length > 1) {
+        component = splitComponent[0];
+        compareValue = splitComponent[1];
+    }
```

Comment:
probably a bit too naive, since you can set up the `key` to be something like `foo.bar` which results in the following submission data:

```json
{"foo": {"bar": "<value>"}}
```

(that's how Formio tranforms it, using `lodash/set` under the hood)

A more robust approach will probably be:

```js
[component, ...compareValueBits] = component.split('.')
```

& then when building the `firstOperand`, you could just `compareValueBits.join('.')` again

Reply:
While updating this, I noticed that my change in the backed also breaks in case of components with dots in the key names. So I made changes there too. I changed a bit approach in the front end